infra/aws/terraform/iam/eks-e2e-boskos/README.md

```diff
@@ -0,0 +1,42 @@
+# IAM for Boskos accounts
+
+This directory contains CloudFormation templates needed to provision IAM resources
```

Comment:
As discussed outside GH, it was done to avoid creating dedicated S3 bucket for TF state.

Reply:
Let me clarify this further for the reference. If we wanted to use Terraform to manage IAM user and roles for these accounts, we would need to:

- Create an IAM user and access key for that user in each account
- Create a S3 bucket in each account to be used for storing Terraform state
- Have a dedicated Terraform backend for each account

In any case, tasks 1 and 2 must be done manually or with CloudFormation. That said, I think that completely managing IAM with CloudFormation is easier and more scalable.

Reply:
I wouldn't put these 3 things into the main Terraform:

1. S3 bucket for Terraform state (do we need one per account, though?)
2. lock table for Terraform state locking
3. IAM role for executing Terraform

Reply:
Also, we could use StackSets if we do want CloudFormation deployed to more than one account.

Reply:
> S3 bucket for Terraform state (do we need one per account, though?)

No, the way it's working right now is that it's enough to have bucket in only one account (in this case boskos-001). With IAM user in that account, we assume roles in other accounts and do the magic. But state for all those accounts is stored in a single bucket.

Reply:
> state for all those accounts is stored in a single bucket

I think that's OK. We can use a bucket policy if we need to restrict access to state, and that means we don't need to make more buckets.

Reply:
> Also, we could use StackSets if we do want CloudFormation deployed to more than one account.

@sftim I looked into StackSets and my understanding is that basically it allows applying CloudFormation in more than one account. That requires creating a role in each account or integrating with AWS Organization. I like that idea, but I'd leave it for a potential follow up, maybe it would be better to have a StackSet in the management account and then manage boskos accounts from there. However, we currently don't have access to the management account.

Reply:
You _can_ delegate access, but it's not fun. Let's rule out StackSets.

Reply:
The reasoning is now documented in README for IAM CloudFormation.